--- if.include	2014-04-08 14:49:58.379248949 +0300
+++ /root/if.include.orig	2014-04-02 10:21:07.000000000 +0300
@@ -94,7 +94,7 @@
         return ifIndex
 
 
-def inventory_if_common(info, type_required = False):
+def inventory_if_common(info):
     if len(info) == 0 or len(info[0]) != 20:
         return []
 
@@ -156,9 +156,6 @@
 
             if ifType in porttypes and ifOperStatus in portstates:
                 params = {}
-                if type_required:
-                    params["iftype"] = ifType
-
                 if if_inventory_monitor_state:
                     params["state"] = [ifOperStatus]
 
@@ -185,33 +182,14 @@
                 params["speed"] = total_speed
             pre_inventory.append( (group_name, "%r" % params, int(ifIndex)) )
 
-    if type_required:
-        inventory = {}
-    else:
-        inventory = []
-
-    import pprint
+    inventory = []
     # Check for duplicate items (e.g. when using Alias as item and the alias is not unique)
     for item, params, index in pre_inventory:
         if item in duplicate:
             new_item = "%s %d" % (item, index)
         else:
             new_item = item
-
-        if type_required:
-            init_params = eval(params)
-            if init_params.has_key("aggregate"):
-                ifType = "_agg"
-            else:
-                ifType = str(init_params["iftype"])
-                del init_params["iftype"]
-
-            if not inventory.has_key(ifType):
-                inventory[ifType] = []
-
-            inventory[ifType].append((new_item, "%r" % init_params))
-        else:
-            inventory.append((new_item, params))
+        inventory.append((new_item, params))
 
     return inventory
 
