--- .check_mk_templates.cfg	2015-05-18 16:02:29.004000000 +0300
+++ check_mk_templates.cfg	2015-05-18 16:03:26.364000000 +0300
@@ -171,8 +171,8 @@
   retain_nonstatus_information    1
   notification_interval           0
   is_volatile                     0
-  normal_check_interval           1
-  retry_check_interval            1
+  check_interval                1
+  retry_interval                1
   max_check_attempts              1
   notification_options            u,c,w,r,f,s
   notification_period             24X7
--- modules/.check_mk.py	2015-05-18 16:03:54.644000000 +0300
+++ modules/check_mk.py	2015-05-18 16:04:27.808000000 +0300
@@ -2322,8 +2322,8 @@
 define service {
   use\t\t\t\t%s
   host_name\t\t\t%s
-  normal_check_interval\t\t%d
-  retry_check_interval\t\t%d
+  check_interval\t\t%d
+  retry_interval\t\t%d
 %s  service_description\t\t%s
 }
 """ % (inventory_check_template, hostname, inventory_check_interval,
