--- check_mk.py	2014-03-20 17:19:35.463249225 +0200
+++ check_mk.py.bkp	2014-03-20 17:19:07.259248947 +0200
@@ -89,6 +89,17 @@
 perfometers["check_mk-ucd_cpu_util"] = perfometer_check_mk_kernel_util
 perfometers["check_mk-lparstat_aix.cpu_util"] = perfometer_check_mk_kernel_util
 
+def perfometer_check_mk_cisco_chassis_load(row, check_command, perf_data):
+    return perfometer_bandwidth(
+            savefloat(perf_data[0][1]), 
+            savefloat(perf_data[1][1]),
+            100, 
+            100, 
+            ''
+        )
+
+perfometers["check_mk-cisco_chassis_load"] = perfometer_check_mk_cisco_chassis_load
+
 def perfometer_check_mk_mem_used(row, check_command, perf_data):
     h = '<table><tr>'
     ram_total  = float(perf_data[0][6])
@@ -302,6 +313,7 @@
 
 perfometers["check_mk-if"] = perfometer_check_mk_if
 perfometers["check_mk-if64"] = perfometer_check_mk_if
+perfometers["check_mk-if64_snu"] = perfometer_check_mk_if
 perfometers["check_mk-vms_if"] = perfometer_check_mk_if
 perfometers["check_mk-if_lancom"] = perfometer_check_mk_if
 perfometers["check_mk-lnx_if"] = perfometer_check_mk_if
